infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/VcsSshKeysProvisioner.java

```diff
@@ -0,0 +1,219 @@
+ *
+ * @author Vitalii Parfonov
+ */
+public class VcsSshKeysProvisioner implements ConfigurationProvisioner<KubernetesEnvironment> {
+
+  // SecretTypeSSHAuth contains data needed for SSH authentication.
+  // Required field:
+  // - Secret.Data["ssh-privatekey"] - private SSH key needed for authentication
+  private static final String SECRET_TYPE_SSH = "kubernetes.io/ssh-auth";
+
+  // SSHAuthPrivateKey is the key of the required SSH private key for SecretTypeSSHAuth secrets
+  private static final String SSH_PRIVATE_KEY = "ssh-privatekey";
+
+  private static final String SSH_BASE_CONFIG_PATH = "/etc/ssh/";
+
+  private static final String SSH_CONFIG = "ssh_config";
+
+  private static final String SSH_CONFIG_PATH = SSH_BASE_CONFIG_PATH + SSH_CONFIG;
+
+  private static final Logger LOG = LoggerFactory.getLogger(VcsSshKeysProvisioner.class);
+
+  private final SshManager sshManager;
+
+  @Inject
+  public VcsSshKeysProvisioner(SshManager sshManager) {
+    this.sshManager = sshManager;
+  }
+
+  @Override
+  @Traced
+  public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
+      throws InfrastructureException {
+
+    TracingTags.WORKSPACE_ID.set(identity::getWorkspaceId);
+
+    StringBuilder sshConfig = new StringBuilder();
+    Map<String, String> map = new HashMap<>();
+
+    try {
+      List<SshPairImpl> sshPairs = sshManager.getPairs(identity.getOwnerId(), "vcs");
+      if (!sshPairs.isEmpty()) {
+        for (SshPair sshPair : sshPairs) {
+          if (isNullOrEmpty(sshPair.getName()) || isNullOrEmpty(sshPair.getPrivateKey())) {
+            continue;
+          }
+          Secret secret =
+              new SecretBuilder()
+                  .addToData(
+                      SSH_PRIVATE_KEY,
+                      Base64.getEncoder().encodeToString(sshPair.getPrivateKey().getBytes()))
+                  .withType(SECRET_TYPE_SSH)
+                  .withNewMetadata()
+                  .withName(getValidNameForSecret(sshPair.getName()))
+                  .endMetadata()
+                  .build();
+
+          k8sEnv.getSecrets().put(secret.getMetadata().getName(), secret);
+
+          if (!sshPairs.isEmpty()) {
+            k8sEnv
+                .getPodsData()
+                .values()
+                .forEach(p -> addSshKeySecret(secret.getMetadata().getName(), p.getSpec()));
+          }
+          sshConfig.append(buildConfig(sshPair.getName()));
+        }
+
+        map.put(SSH_CONFIG, sshConfig.toString());
+        ConfigMap configMap =
+            new ConfigMapBuilder()
+                .withNewMetadata()
+                .withName("sshconfigmap")
+                .endMetadata()
+                .withData(map)
+                .build();
+
+        k8sEnv.getConfigMaps().put(configMap.getMetadata().getName(), configMap);
+        k8sEnv.getPodsData().values().forEach(p -> addConfigFile(p.getSpec()));
+      }
+    } catch (ServerException e) {
+      LOG.warn("Unable get SSH Keys ", e);
```

Comment:
Maybe it makes sense to explain message a bit

Reply:
+1. Something like 
```
 LOG.warn("Unable get SSH Keys. Cause: %s", e.getMessage(), e);
```

Reply:
done